pkg/mcproducer/scripts/replace-kernel-module.sh

```diff
@@ -19,17 +21,22 @@ EOF
     cat "$kmm_config_file_filepath"
 }
 
-echo "before checking image tar file presence"
-if [ -e $kernel_module_image_filepath ]; then
-    echo "Image file $kernel_module_image_filepath found on the local file system, creating kmm config file"
+echo "before checking image presence"
+if [ -n "$(podman images -q $full_kernel_module_image 2> /dev/null)" ]; then
+    echo "Image $worker_image found on the local file system, creating kmm config file"
     create_kmm_config
-    echo "running kernel-management worker image"
-    podman run --user=root --privileged -v /lib/modules:/lib/modules -v $kmm_config_file_filepath:/etc/kmm-worker/config.yaml -v $kernel_module_image_filepath:$kernel_module_image_filepath $worker_image kmod load --tarball /etc/kmm-worker/config.yaml
+    podman pod create --name kmm-pod
+    echo "running init container image"
+    init_container_id=$(podman create --pod kmm-pod --user=root --init-ctr=once --privileged -v /tmp:/tmp $full_kernel_module_image /bin/sh -c "mkdir -p /tmp/opt/lib/modules; cp -R /opt/lib/modules/* /tmp/opt/lib/modules;")
```

Comment:
Don't we also need to copy firmware files?
Why is this container has to be `privileged`?

Reply:
@ybettan why should we copy firmware files? they are part of the container image and will be dealt by init-container

Reply:
@ybettan are you sure?
because if I remove` --privileged` I get an error of `mkdir: cannot create directory '/tmp/opt': Permission denied`

Reply:
> they are part of the container image and will be dealt by init-container

@yevgeny-shnaidman the init-container won't do anything with them unless it is told to like @TomerNewman did with the modules files when overriding the entrypoint of the container.

Am I missing something?

Reply:
> @ybettan are you sure?
because if I remove --privileged I get an error of mkdir: cannot create directory '/tmp/opt': Permission denied

I am not sure, but in day 2 only the worker container is privileged - https://github.com/kubernetes-sigs/kernel-module-management/blob/0e806c0d0757e178a001ec8d62e912f5fcdd0a0a/internal/controllers/nmc_reconciler.go#L1144.

Reply:
@ybettan yes, you are right, i have missed that part

Reply:
@ybettan regarding the `--privileged` flag, I removed it.
and following our slack conversation, we will copy the firmware files in the next pr.